device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -401,6 +401,18 @@ public void addMessage(Message message, IotHubEventCallback callback, Object cal
 
         //Codes_SRS_IOTHUBTRANSPORT_34_042: [This function shall build a transport packet from the provided message,
         // callback, and context and then add that packet to the waiting queue.]
+
+        if (message instanceof BatchMessages && !(this.iotHubTransportConnection instanceof HttpsIotHubConnection))
```

Comment:
If you want to allow amqp/mqtt to use this batch API, then at least add a comment here explaining why only https doesn't split the batch message here